staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go

```diff
@@ -213,6 +215,32 @@ func newAuthenticator(c *cache, config *api.ExecConfig, cluster *clientauthentic
 	return c.put(key, a), nil
 }
 
+func isInteractive(isTerminalFunc func(int) bool, config *api.ExecConfig, stdinUnavailable bool) (bool, error) {
+	var shouldBeInteractive bool
+	switch config.InteractiveMode {
+	case "":
+		// backwards compatible default
+		shouldBeInteractive = isTerminalFunc(int(os.Stdin.Fd()))
+	case api.NeverExecInteractiveMode:
+		shouldBeInteractive = false
+	case api.IfAvailableExecInteractiveMode:
+		shouldBeInteractive = isTerminalFunc(int(os.Stdin.Fd()))
+	case api.AlwaysExecInteractiveMode:
+		if !isTerminalFunc(int(os.Stdin.Fd())) {
+			return false, errors.New("standard input is not a terminal")
+		}
+		shouldBeInteractive = true
+	default:
+		return false, fmt.Errorf("unknown ExecInteractiveMode: %q", config.InteractiveMode)
+	}
+
+	if shouldBeInteractive && stdinUnavailable {
+		return false, fmt.Errorf("standard input is unavailable")
+	}
```

Comment:
Is this correct for when `mode=IfAvailableExecInteractiveMode`, `shouldBeInteractive=true` and `stdinUnavailable=true`?

Reply:
This should now be correct, and read a little bit better, for when `mode=IfAvailableExecInteractiveMode` and `isTerminalFunc()` returns `true` thanks to 76281c20ccd.